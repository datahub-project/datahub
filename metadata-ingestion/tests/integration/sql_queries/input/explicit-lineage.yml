source:
  type: sql-queries
  config:
    query_file: ./input/explicit-lineage.jsonl
    platform: snowflake
    use_schema_resolver: false
    usage:
      # Values to match timestamps in input queries file (Jan 1, 2021)
      start_time: 2021-01-01T00:00:00Z
      end_time: 2021-01-02T00:00:00Z
      bucket_duration: DAY

sink:
  type: file
  config:
    filename: ./output.json

datahub_api:
  server: http://localhost:8080