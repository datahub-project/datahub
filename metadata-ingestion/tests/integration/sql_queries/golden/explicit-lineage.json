[
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.audit_log,PROD)",
    "changeType": "UPSERT",
    "aspectName": "upstreamLineage",
    "aspect": {
        "json": {
            "upstreams": [
                {
                    "auditStamp": {
                        "time": 1753276219583,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "created": {
                        "time": 1609459260000,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "dataset": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
                    "type": "TRANSFORMED",
                    "query": "urn:li:query:1cfc42a2716bb27fa618416438130aedfaa59279874ae462e33124070d61433c"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219583,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:1cfc42a2716bb27fa618416438130aedfaa59279874ae462e33124070d61433c",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "INSERT INTO snowflake.db.orders SELECT user_id, product_id, order_date FROM snowflake.db.temp_orders",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459260000,
                "actor": "urn:li:corpuser:_ingestion"
            },
            "lastModified": {
                "time": 1609459260000,
                "actor": "urn:li:corpuser:_ingestion"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219584,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:1cfc42a2716bb27fa618416438130aedfaa59279874ae462e33124070d61433c",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                },
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.audit_log,PROD)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219585,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:1cfc42a2716bb27fa618416438130aedfaa59279874ae462e33124070d61433c",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219585,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.daily_summary,PROD)",
    "changeType": "UPSERT",
    "aspectName": "upstreamLineage",
    "aspect": {
        "json": {
            "upstreams": [
                {
                    "auditStamp": {
                        "time": 1753276219586,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "created": {
                        "time": 1609459320000,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "dataset": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.sessions,PROD)",
                    "type": "TRANSFORMED",
                    "query": "urn:li:query:d0bbe413da3060e9fddc8ba3aa139f657e62bf9bbcfa2b5969e1fc305e773ee2"
                },
                {
                    "auditStamp": {
                        "time": 1753276219586,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "created": {
                        "time": 1609459320000,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "dataset": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.shipments,PROD)",
                    "type": "TRANSFORMED",
                    "query": "urn:li:query:d0bbe413da3060e9fddc8ba3aa139f657e62bf9bbcfa2b5969e1fc305e773ee2"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219586,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:d0bbe413da3060e9fddc8ba3aa139f657e62bf9bbcfa2b5969e1fc305e773ee2",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "CREATE VIEW snowflake.db.user_summary AS SELECT u.id, u.name, COUNT(o.id) as order_count FROM snowflake.db.users u LEFT JOIN snowflake.db.orders o ON u.id = o.user_id GROUP BY u.id, u.name",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459320000,
                "actor": "urn:li:corpuser:_ingestion"
            },
            "lastModified": {
                "time": 1609459320000,
                "actor": "urn:li:corpuser:_ingestion"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219587,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:d0bbe413da3060e9fddc8ba3aa139f657e62bf9bbcfa2b5969e1fc305e773ee2",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.sessions,PROD)"
                },
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.shipments,PROD)"
                },
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.daily_summary,PROD)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219587,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:d0bbe413da3060e9fddc8ba3aa139f657e62bf9bbcfa2b5969e1fc305e773ee2",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219588,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.customers,PROD)",
    "changeType": "UPSERT",
    "aspectName": "datasetUsageStatistics",
    "aspect": {
        "json": {
            "timestampMillis": 1609459200000,
            "eventGranularity": {
                "unit": "DAY",
                "multiple": 1
            },
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "uniqueUserCount": 1,
            "totalSqlQueries": 1,
            "topSqlQueries": [
                "SELECT COUNT(*) FROM snowflake.db.customers WHERE status = 'active'"
            ],
            "userCounts": [
                {
                    "user": "urn:li:corpuser:analyst",
                    "count": 1
                }
            ],
            "fieldCounts": []
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974459,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.audit_log,PROD)",
    "changeType": "UPSERT",
    "aspectName": "operation",
    "aspect": {
        "json": {
            "timestampMillis": 1753276219588,
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "operationType": "UNKNOWN",
            "sourceType": "DATA_PLATFORM",
            "lastUpdatedTimestamp": 1609459260000,
            "queries": [
                "urn:li:query:1cfc42a2716bb27fa618416438130aedfaa59279874ae462e33124070d61433c"
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219588,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.daily_summary,PROD)",
    "changeType": "UPSERT",
    "aspectName": "operation",
    "aspect": {
        "json": {
            "timestampMillis": 1753276219589,
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "operationType": "UNKNOWN",
            "sourceType": "DATA_PLATFORM",
            "lastUpdatedTimestamp": 1609459320000,
            "queries": [
                "urn:li:query:d0bbe413da3060e9fddc8ba3aa139f657e62bf9bbcfa2b5969e1fc305e773ee2"
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753276219589,
        "runId": "sql-queries-2025_07_23-15_10_19-kfua9t",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.metrics,PROD)",
    "changeType": "UPSERT",
    "aspectName": "operation",
    "aspect": {
        "json": {
            "timestampMillis": 1753430974462,
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "actor": "urn:li:corpuser:etl_job",
            "operationType": "INSERT",
            "sourceType": "DATA_PLATFORM",
            "lastUpdatedTimestamp": 1609459440000,
            "queries": [
                "urn:li:query:9519c8bdff1e9bb71ea82da930105c5e6462bbb031ea026f714e529596c610f4"
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974462,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9519c8bdff1e9bb71ea82da930105c5e6462bbb031ea026f714e529596c610f4",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.metrics,PROD)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974463,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9519c8bdff1e9bb71ea82da930105c5e6462bbb031ea026f714e529596c610f4",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "INSERT INTO snowflake.db.metrics (date, count) VALUES (CURRENT_DATE, 42)",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459440000,
                "actor": "urn:li:corpuser:etl_job"
            },
            "lastModified": {
                "time": 1609459440000,
                "actor": "urn:li:corpuser:etl_job"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974462,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9519c8bdff1e9bb71ea82da930105c5e6462bbb031ea026f714e529596c610f4",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974463,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9519c8bdff1e9bb71ea82da930105c5e6462bbb031ea026f714e529596c610f4",
    "changeType": "UPSERT",
    "aspectName": "queryUsageStatistics",
    "aspect": {
        "json": {
            "timestampMillis": 1609459200000,
            "eventGranularity": {
                "unit": "DAY",
                "multiple": 1
            },
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "queryCount": 1,
            "uniqueUserCount": 1,
            "userCounts": [
                {
                    "user": "urn:li:corpuser:etl_job",
                    "count": 1
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974464,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9105218d713755517e46ea672d2a74d6161b421fc58ba9653126d326681aa091",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.customers,PROD)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974465,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9105218d713755517e46ea672d2a74d6161b421fc58ba9653126d326681aa091",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "SELECT COUNT(*) FROM snowflake.db.customers WHERE status = 'active'",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459380000,
                "actor": "urn:li:corpuser:analyst"
            },
            "lastModified": {
                "time": 1609459380000,
                "actor": "urn:li:corpuser:analyst"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974465,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9105218d713755517e46ea672d2a74d6161b421fc58ba9653126d326681aa091",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974466,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:9105218d713755517e46ea672d2a74d6161b421fc58ba9653126d326681aa091",
    "changeType": "UPSERT",
    "aspectName": "queryUsageStatistics",
    "aspect": {
        "json": {
            "timestampMillis": 1609459200000,
            "eventGranularity": {
                "unit": "DAY",
                "multiple": 1
            },
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "queryCount": 1,
            "uniqueUserCount": 1,
            "userCounts": [
                {
                    "user": "urn:li:corpuser:analyst",
                    "count": 1
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1753430974466,
        "runId": "sql-queries-2025_07_25-10_09_34-9ej8ym",
        "lastRunId": "no-run-id-provided"
    }
}
]